internal/healthchecks/api_check.go

```diff
@@ -97,18 +93,13 @@ func createMonitoringPingRequest(gceMetadata resourcedetector.GCEResource) *moni
 // time series point with empty values to an Ops Agent specific metric.
 // This method mirrors the "(c *Client) Ping" method in "cloud.google.com/go/logging".
 func monitoringPing(ctx context.Context, client monitoring.MetricClient, gceMetadata resourcedetector.GCEResource) error {
-	var err error
-	for i := 0; i < MaxMonitoringPingRetries; i++ {
-		err = client.CreateTimeSeries(ctx, createMonitoringPingRequest(gceMetadata))
-		if err == nil || !isInvalidArgumentErr(err) { 
-			break
-		}
-		// This fixes b/291631906 when the monitoringPing is retried very quickly resulting
-		// in an `InvalidArgument` error due a maximum write rate of one point every 5 seconds.
-		// https://cloud.google.com/monitoring/quotas 
-		time.Sleep(6 * time.Second)
-	}
-	return err
+	// Points written to a time series must be at least 5 seconds apart. Because `monitoringPing` might
+	// be called multiple times in quick succession, the first attempted request to `CreateTimeSeries`
+	// may fail. We can retry the request >5 seconds later in such cases.
+	// https://cloud.google.com/monitoring/quotas
+	pingBackoff := backoff.WithMaxRetries(backoff.NewConstantBackOff(6*time.Second), MaxMonitoringPingRetries)
+	pingOperation := func() error { return client.CreateTimeSeries(ctx, createMonitoringPingRequest(gceMetadata)) }
```

Comment:
@franciscovalentecastro Am I missing something, or will we now also retry `InvalidArgument` errors?

Reply:
See https://github.com/GoogleCloudPlatform/ops-agent/pull/1433#discussion_r1333368553

Reply:
Thanks, missed that.